components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/kafka/KAFKAPollingConsumer.java

```diff
@@ -0,0 +1,171 @@
+    private long scanInterval;
+    private Long lastRanTime;
+
+    /**
+     * Initialize the kafka properties and the polling interval
+     */
+    public KAFKAPollingConsumer(Properties kafkaProperties, long interval) throws Exception {
+
+        this.kafkaProperties = kafkaProperties;
+        this.scanInterval = interval;
+        try {
+            if (kafkaProperties.getProperty(KAFKAConstants.THREAD_COUNT) == null
+                    || kafkaProperties.getProperty(KAFKAConstants.THREAD_COUNT)
+                    .equals("")
+                    || Integer.parseInt(kafkaProperties
+                    .getProperty(KAFKAConstants.THREAD_COUNT)) <= 0) {
+                this.threadCount = 1;
+            } else {
+                this.threadCount = Integer.parseInt(kafkaProperties
+                        .getProperty(KAFKAConstants.THREAD_COUNT));
+            }
+        } catch (NumberFormatException nfe) {
+            logger.error("Invalid numeric value for thread count.");
+            throw new SynapseException("Invalid numeric value for thread count.", nfe);
+        }
+        if (kafkaProperties.getProperty(KAFKAConstants.TOPICS) != null) {
+            this.topics = Arrays.asList(kafkaProperties.getProperty(
+                    KAFKAConstants.TOPICS).split(","));
+        }
+    }
+
+    /**
+     * Start the listener to listen when new messages come to the esb,the listener can be high level or low level.
+     */
+    public void startsMessageListener() throws Exception {
+
+        logger.debug("Create the Kafka message listener");
+        if (messageListener == null) {
+            //Start a high level listener
+            try {
+                if (kafkaProperties.getProperty(KAFKAConstants.CONSUMER_TYPE) == null
+                        || kafkaProperties
+                        .getProperty(KAFKAConstants.CONSUMER_TYPE)
+                        .isEmpty()
+                        || kafkaProperties
+                        .getProperty(KAFKAConstants.CONSUMER_TYPE)
+                        .equalsIgnoreCase(
+                                AbstractKafkaMessageListener.CONSUMER_TYPE.HIGHLEVEL
+                                        .getName())) {
+                    messageListener = new KAFKAMessageListener(threadCount, topics,
+                            kafkaProperties, injectHandler);
+                    //Start a low level listener
+                } else if (kafkaProperties
+                        .getProperty(KAFKAConstants.CONSUMER_TYPE)
+                        .equalsIgnoreCase(
+                                AbstractKafkaMessageListener.CONSUMER_TYPE.SIMPLE
+                                        .getName())) {
+                    messageListener = new SimpleKafkaMessageListener(
+                            kafkaProperties, injectHandler);
+                }
+            } catch (Exception e) {
+                logger.error("The consumer type should be high level or simple");
```

Comment:
There are no specific exceptions in the above code. I will check and if the generic exception is not necessary I will remove.